src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -506,6 +506,8 @@ public function resolveConcatType(Type $left, Type $right): Type
 
 		$leftNumericStringNonEmpty = TypeCombinator::remove($leftStringType, new ConstantStringType(''));
 		if ($leftNumericStringNonEmpty->isNumericString()->yes()) {
+			$allowedPattern = $left->isInteger()->yes() ? '#^[0-9.]+$#' : '#^[0-9]+$#';
```

Comment:
```suggestion
			$allowedRightPattern = $left->isInteger()->yes() ? '#^[0-9.]+$#' : '#^[0-9]+$#';
```

Reply:
Wouldn't it be kinda weird having those two private constant in InitializerExprTypeResolver with 2000+ unrelated lines of code.

Reply:
you could also put the information as a comment on the patterns